core/src/main/java/org/opensearch/sql/expression/aggregation/CountAggregator.java

```diff
@@ -50,7 +52,7 @@ public CountAggregator(List<Expression> arguments, ExprCoreType returnType) {
 
   @Override
   protected CountState iterate(ExprValue value, CountState state) {
-    state.count++;
+    state.count(value, distinct);
```

Comment:
Can the `distinct` be passed once when creating state object in `create()`? If so, I'm thinking how about we add another `DistinctCountState` to avoid adding complexity to existing `CountState`?

Reply:
Done, thanks!